### ISSUE TITLE
No Matrix server available with good latency, requests takes more than 1.0 seconds.

### ISSUE THREAD

**Issue by agatsoh** [raiden-network/raiden#5626]
## Problem Definition

PFS2 node2 stdout gives matrix error

```
Checking if the ethereum node is synchronized
You have chosen the Pathfinding Service at https://pfs-goerli-with-fee.services-dev.raiden.network.
Operator: John Doe, running version: 0.5.0, chain_id: 5.
Fees will be paid to 0x062C12c01D0f17fC9eAa33940D994594d91a0182. Each request costs 1e-16 RDN.
Message from the Pathfinding Service:
This is your favorite pfs for token network registry 0x62436bbadB629498107278486C2A367FfdFEb2f3
Raiden is running in development mode

You are connected to the 'goerli' network and the DB path is: /data/scenarios/pfs2_simple_no_path/node_94_002/node_c76bc8e0/netid_5/network_62436bba/v25_log.db
FATAL: An unexpected exception occured. A traceback has been written to /tmp/raiden-exception-2020-01-10T01-09o4cdarhn.txt
No Matrix server available with good latency, requests takes more than 1.0 seconds.
```

**No Matrix server available with good latency, requests takes more than 1.0 seconds.**

logs
https://drive.google.com/drive/folders/1MdROjSbLLifDGI_oiuVdFEN_N1eDxWAs?usp=sharing

**Comment by Dominik1999** [comment]
We have seen that more often than once

**Comment by agatsoh** [comment]
This exception occured again with individual nodes in PFS4 scenario. The logs have been attached below.
[pfs4_use_best_path_13_01_2020.tar.gz](https://github.com/raiden-network/raiden/files/4052636/pfs4_use_best_path_13_01_2020.tar.gz)

**Comment by ulope** [comment]
Ok I think I know what's happening here.

The server selection behaviour was changed in #5544 to only wait for the fastest greenlet to return.
The round trip time is sampled three times. Unfortunately if for any server the first or second request fails this greenlet will in almost all cases be the first one to return.

Additionally the 1s request timeout seems to be a bit too short. I performed some tests and even moderate amounts of network or event loop congestion can push the requests over that limit.
